onnxruntime/test/onnx/microbenchmark/tptest.cc

```diff
@@ -35,7 +35,7 @@ BENCHMARK(BM_CreateThreadPool)
 #pragma GCC optimize("O0")
 #endif
 void SimpleForLoop(ptrdiff_t first, ptrdiff_t last) {
-  size_t sum = 0;
+  [[maybe_unused]] size_t sum = 0;
```

Comment:
if the compiler is warning about it, could it also be "smart" enough to optimize it away? should we also do something similar to the other change in the PR with `benchmark::DoNotOptimize()`?